keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow.keras.__internal__.layers import BaseRandomLayer
+
+from keras_cv.utils import conv_utils
+
+
+class DropBlock2D(BaseRandomLayer):
+    """Applies DropBlock regularization to input features.
+
+    DropBlock is a form of structured dropout, where units in a contiguous
+    region of a feature map are dropped together. DropBlock works better than
+    dropout on convolutional layers due to the fact that activation units in
+    convolutional layers are spatially correlated.
+
+    It is advised to use DropBlock after activation in Conv -> BatchNorm -> Activation
+    block in further layers of the network. For example, the paper mentions using
+    DropBlock in 3rd and 4th group of ResNet blocks.
+
+    Reference:
+    - [DropBlock: A regularization method for convolutional networks](
+        https://arxiv.org/abs/1810.12890
+    )
+
+    Args:
+        keep_probability: float. Probability of keeping a unit. Defaults to 0.9.
+            Must be between 0 and 1. For best results, the value should be between
+            0.75-0.95
+        dropblock_size: integer. The size of the block to be dropped. Defaults to 7.
```

Comment:
Also, should the dropblock shape be a static value? or we can do dynamic shape based on the inputs?

Reply:
The paper mentioned that they always used value 7. I'm not sure if I understand the dynamic shape aspect - could you provide an example?

Reply:
Also, quite a lot of computation could be moved from `call` to the `build` function - almost everything before creating random noise. 
This will speed up the execution, but the user might get garbage results if they reuse the same layer for differently sized input.

Reply:
> Also, should the dropblock shape be a static value? or we can do dynamic shape based on the inputs?
I think what Scott means by this (correct me if I'm wrong) is that perhaps we can support randomly sampled dropblock_size from a range.

so...

```
dropblock_size=(3, 7)
```

would randomly pick from the range 3, 7 in each dropblock invocation.

Reply:
@LukeWood Thanks for the explanation. However, I'm not sure about this option. It is not in the paper and not in the other implementations linked in this PR's comments. It's hard to say how big the impact could be with this option.

Reply:
yeah, its probably overkill.